qcodes/interactive_widget.py

```diff
@@ -0,0 +1,436 @@
+
+    def _button_to_input(dct: Dict, box: Box) -> Callable[[Button], None]:
+        def on_click(_: Button) -> None:
+            description = _yaml_dump(
+                dct
+            )  # TODO: include and extract more data!
+            text_input = Textarea(
+                value=description,
+                placeholder="Enter text",
+                disabled=True,
+                layout=Layout(height="300px", width="auto"),
+            )
+            plot_button = button(
+                "Plot",
+                "warning",
+                on_click=_do_in_tab(tab, ds, "plot"),
+                button_kwargs=dict(icon="line-chart"),
+            )
+            snapshot_button = button(
+                "Open snapshot",
+                "warning",
+                on_click=_do_in_tab(tab, ds, "snapshot"),
+                button_kwargs=dict(icon="camera"),
+            )
+            back_button = button(
+                "Back",
+                "warning",
+                on_click=_input_to_button(dct, box),
+                button_kwargs=dict(icon="undo"),
+            )
+            box.children = (
+                text_input,
+                snapshot_button,
+                plot_button,
+                back_button,
+            )
+
+        return on_click
+
+    def _input_to_button(dct: Dict, box: Box) -> Callable[[Button], None]:
+        def on_click(_: Button) -> None:
+            box.children = (_changeable_button(dct, box),)
+
+        return on_click
+
+    def _changeable_button(dct: Dict, box: Box) -> Button:
+        return button(
+            ", ".join(dct), "success", on_click=_button_to_input(dct, box),
+        )
+
+    box = VBox([], layout=Layout(height="auto", width="auto"))
+    box.children = (_changeable_button(dct, box),)
+    return box
+
+
+def _get_coords_and_vars(ds: DataSet) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    coordinates = {}
+    variables = {}
+    for p, spec in ds.paramspecs.items():
+        attrs = {
+            "unit": spec.unit,
+            "label": spec.label,
+            "type": spec.type,
+        }
+        if spec.depends_on:
+            attrs["depends_on"] = spec.depends_on.split(", ")  # type: ignore
+            variables[p] = attrs
+        else:
+            coordinates[p] = attrs
+    return coordinates, variables
+
+
+def _experiment_widget(tab: Tab) -> GridspecLayout:
+    """Show a `ipywidgets.GridspecLayout` with information about the
+    loaded experiment. The clickable buttons can perform an action in ``tab``.
+    """
+    header_names = [
+        "Run ID",
+        "Name",
+        "Experiment",
+        "Coordinates",
+        "Variables",
+        "MSMT Time",
+        "Notes",
+    ]
+
+    header = {n: button(n, "info") for n in header_names}
+    rows = [header]
+    for exp in qcodes.experiments():
+        tooltip = f"{exp.name}#{exp.sample_name}@{exp.path_to_db}"
```

Comment:
wrt `path_to_db` - depending on the usage of the widget, it might iether make sense to put to the top of the table, or, in case we allow listing arbitrary datasets, path_to_db could be part of that "expandable identity info" that i mentioned in a few other comments.

Reply:
I've added it to the Experiment button in 72e07e1a2271aa2a904569138af4aa02bf7bb34d.